### PR TITLE
config(overlays): reorganized config file

### PR DESCRIPTION
**Pull request by yairtawil**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
 		"e2e": "ng e2e",
 		"fix-packages": "node scripts/fix-packages",
 		"fix-package-json": "node scripts/fix-package-json",
-		"fix-all": "npm run fix-packages && npm run fix-package-json",
+		"fix-all": "npm run fix-packages && npm run fix-package-json && git add -A",
```
Comment (yuvalbl) [inline package.json]: git add should not be part of fix script  / commit hook but rather be used explicitly.

Comment (yairtawil) [reply package.json]: why not?
'fix-packages' script fix your 'imports', but changes isn't includes on the commit.
'pre-commit'  is finished with your '../pakcages/' old mistakes.

File: src/app/app/app-effects/effects/map.app.effects.ts [hunk 2]
```diff
@@ -214,7 +210,8 @@ export class MapAppEffects {
 				.map(layer => {
 					if (overlay.isGeoRegistered) {
 						if (intersection < this.config.overlayCoverage) {
-							communicator.resetView(layer, position, extentFromGeojson(overlay.footprint));
+							const geoJsonExtent = layer.getProperties().geoJsonExtent || overlay.footprint;
```
Comment (asafMasa) [inline src/app/app/app-effects/effects/map.app.effects.ts]: Check "layer.getProperties().geoJsonExtent "

File: src/app/app/app-effects/effects/tools.app.effects.ts [hunk 3]
```diff
@@ -143,6 +143,9 @@ export class ToolsAppEffects {
 			(action: DisplayOverlaySuccessAction, cases: ICasesState, mapsState: IMapState) => {
 				return [action, cloneDeep(cases.selectedCase), mapsState];
 			})
+		.filter(([action, selectedCase, mapsState]: [DisplayOverlaySuccessAction, Case, IMapState]) => {
```
Comment (asafMasa) [inline src/app/app/app-effects/effects/tools.app.effects.ts]: as discussed

File: src/app/app/app-effects/effects/tools.app.effects.ts [hunk 4]
```diff
@@ -234,11 +237,14 @@ export class ToolsAppEffects {
...
-		.withLatestFrom(this.store$.select(casesStateSelector), this.store$.select(mapStateSelector),
-			(action: SetAutoImageProcessing, cases: ICasesState, mapsState: IMapState) => {
-				const activeMap: CaseMapState = MapFacadeService.activeMap(mapsState);
-				return [activeMap, cloneDeep(cases.selectedCase)];
-			})
+		.withLatestFrom(this.store$.select(casesStateSelector), this.store$.select(mapStateSelector))
+		.filter(([action, cases, mapsState]: [DisplayOverlaySuccessAction, ICasesState, IMapState]) => {
+			return !action.payload.mapId || action.payload.mapId === mapsState.activeMapId;
+		})
+		.map(([action, cases, mapsState]: [DisplayOverlaySuccessAction, ICasesState, IMapState]) => {
```
Comment (asafMasa) [inline src/app/app/app-effects/effects/tools.app.effects.ts]: as discussed

File: src/app/app/app-effects/effects/tools.app.effects.ts (hunk 3 above)
Comment (AmitMY) [reply src/app/app/app-effects/effects/tools.app.effects.ts]: not only for active

File: src/app/app/app-effects/effects/tools.app.effects.ts [hunk 5]
```diff
@@ -141,14 +141,12 @@ export class ToolsAppEffects {
 		.ofType<DisplayOverlaySuccessAction>(OverlaysActionTypes.DISPLAY_OVERLAY_SUCCESS)
 		.withLatestFrom(this.store$.select(casesStateSelector), this.store$.select(mapStateSelector),
 			(action: DisplayOverlaySuccessAction, cases: ICasesState, mapsState: IMapState) => {
-				return [action, cloneDeep(cases.selectedCase), mapsState];
+				const mapId = (action.payload && action.payload.mapId) || mapsState.activeMapId;
```
Comment (AmitMY) [inline src/app/app/app-effects/effects/tools.app.effects.ts]: no

File: src/app/app/app-effects/effects/tools.app.effects.ts [hunk 6]
```diff
@@ -238,17 +236,15 @@ export class ToolsAppEffects {
...
-			return [activeMap, cloneDeep(cases.selectedCase)];
+			const mapId = (action.payload && action.payload.mapId) || mapsState.activeMapId;
+			const selectedMap: CaseMapState = MapFacadeService.mapById(mapsState.mapsList, mapId);
+			return [selectedMap, cloneDeep(cases.selectedCase)];
 		})
-		.filter(([activeMap]: [CaseMapState, Case]) => Boolean(activeMap.data.overlay))
-		.filter(([activeMap, selectedCase]: [CaseMapState, Case]) => {
-			const overlayId = activeMap.data.overlay.id;
-			const isAutoProcessOn = activeMap.data.isAutoImageProcessingActive;
+		.filter(([selectedMap]: [CaseMapState, Case]) => Boolean(selectedMap) && Boolean(selectedMap.data.overlay))
```
Comment (AmitMY) [inline src/app/app/app-effects/effects/tools.app.effects.ts]: boolean

**Pull request merged**